### PR TITLE
Add automatic generation of tensors for Onnxruntime Perf Runner

### PR DESCRIPTION
**Pull request by ryanlai2**
**Motivation and Context**

This change introduces an optional flag to generate arbitrary input data for a model to collect performance. This is an incremental change to the tool to allow generation of input buffers for models with tensor input (maps and sequences should go in different PR).

- Why is this change required? What problem does it solve?
There is a scenario where people have an onnx model and want to gather performance but doesn't have input data in protobuf format yet.  With this change, users of this tool won't need to provide an input data file when collecting performance if this optional flag is provided.

### CODE REVIEW

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 1]
```diff
@@ -136,5 +136,68 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
...
+            break;
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT64:
+            byteCount *= sizeof(uint64_t);
+            break;
+          default:
+            printf("Generation of input for this element type not implemented.");
+            return false;
+        }
+        void* input_tensor_values = malloc(byteCount);
+        generated_input_buffers_.push_back(input_tensor_values);
```
Comment (tiagoshibata) [inline onnxruntime/test/perftest/ort_test_session.cc:194]: if generated_input_buffers_ were a `std::vector<std::vector<int8_t>> generated_input_buffers_`, you could replace these lines with `generated_input_buffers_.emplace_back(byteCount)` and enjoy RAII by removing the explicit `free` calls later on :)

Comment (ryanlai2) [reply onnxruntime/test/perftest/ort_test_session.cc:194]: I'm not so sure how this works 😅

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 2]
```diff
@@ -136,5 +136,68 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
...
+    if (typeInfo.GetONNXType() == ONNX_TYPE_TENSOR) {
+        auto tensorInfo = typeInfo.GetTensorTypeAndShapeInfo();
+        std::vector<int64_t> input_node_dim = tensorInfo.GetShape();
+        int64_t input_tensor_size = 1;
+        for (auto dim : input_node_dim)
+        {
+          input_tensor_size *= dim;
+        }
+        size_t byteCount = input_tensor_size;
+        switch (tensorInfo.GetElementType())
```
Comment (jeffbloo) [inline onnxruntime/test/perftest/ort_test_session.cc:157]: Consider a helper method to get element size.  Is there not one already?

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 3]
```diff
@@ -136,5 +136,68 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
...
+  size_t num_input_nodes;
+  num_input_nodes = session_.GetInputCount();
+
+  // iterate over all input nodes
+  for (size_t i = 0; i < num_input_nodes; i++) {
+    Ort::TypeInfo typeInfo = session_.GetInputTypeInfo(i);
+    Ort::MemoryInfo memory_info = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
+    if (typeInfo.GetONNXType() == ONNX_TYPE_TENSOR) {
+        auto tensorInfo = typeInfo.GetTensorTypeAndShapeInfo();
+        std::vector<int64_t> input_node_dim = tensorInfo.GetShape();
```
Comment (jeffbloo) [inline onnxruntime/test/perftest/ort_test_session.cc:150]: How does this handle free dimensions?  I assume the intention is to convert them to 1s like Winmlrunner?

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 4]
```diff
@@ -136,5 +136,68 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
...
+        int64_t input_tensor_size = 1;
+        for (auto dim : input_node_dim)
+        {
+          input_tensor_size *= dim;
+        }
+        size_t byteCount = input_tensor_size;
+        switch (tensorInfo.GetElementType())
+        {
+          case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT:
```
Comment (jeffbloo) [inline onnxruntime/test/perftest/ort_test_session.cc:159]: Float16 and others are missing

File: onnxruntime/test/perftest/ort_test_session.cc (hunk 3 above)
Comment (ryanlai2) [reply onnxruntime/test/perftest/ort_test_session.cc]: Oh right, this current implementation doesn't handle free dimensions. Let me look into it more

File: onnxruntime/test/perftest/ort_test_session.cc (hunk 1 above)
Comment (ryanlai2) [reply onnxruntime/test/perftest/ort_test_session.cc]: Nevermind, I'm migrating away from this malloc approach :)

File: onnxruntime/test/perftest/ort_test_session.cc (hunk 2 above)
Comment (ryanlai2) [reply onnxruntime/test/perftest/ort_test_session.cc]: No need when using allocators. ORT will handle it internally.

File: onnxruntime/test/perftest/ort_test_session.cc (hunk 3 above)
Comment (ryanlai2) [reply onnxruntime/test/perftest/ort_test_session.cc]: I will treat free dimensions as 1.

File: onnxruntime/test/perftest/ort_test_session.cc (hunk 4 above)
Comment (ryanlai2) [reply onnxruntime/test/perftest/ort_test_session.cc]: No need when using allocators. ORT will handle it internally with tensor_info.GetElementType().

**Pull request merged**